File: Android/Android/app/src/main/java/com/copanostudios/marylanddailytrivia/data/TriviaQuestion.kt
package com.copanostudios.marylanddailytrivia.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class TriviaDifficulty {
    @SerialName("easy") EASY,
    @SerialName("medium") MEDIUM,
    @SerialName("hard") HARD
}

@Serializable
data class TriviaQuestion(
    val id: String,
    val category: String,
    val difficulty: TriviaDifficulty,
    val question: String,
    val choices: List<String>,
    val correctIndex: Int,
    val explanation: String? = null,
    val source: String? = null,
    val tags: List<String>? = null
) {
    val isValid: Boolean
        get() = choices.size == 4 &&
                correctIndex in 0 until choices.size &&
                question.isNotBlank()
}

File: iOS/Maryland Daily Trivia/Maryland Daily Trivia/TriviaQuestion.swift
//
//  TriviaQuestion.swift
//  Maryland Daily Trivia
//
//  Created by Ramon Dominguez on 1/1/26.
//
import Foundation

enum TriviaDifficulty: String, Codable, CaseIterable {
    case easy, medium, hard
}

struct TriviaQuestion: Codable, Identifiable, Hashable {
    let id: String                  // stable unique id, e.g. "tx_hist_0001"
    let category: String            // "History", "Geography", "Sports", etc.
    let difficulty: TriviaDifficulty
    let question: String
    let choices: [String]           // 4 choices
    let correctIndex: Int           // 0...3
    let explanation: String?        // shown after answer (optional)

    // Optional metadata (safe to ignore in UI for now)
    let source: String?             // internal note / citation
    let tags: [String]?             // e.g. ["alamo", "san-antonio"]

    var isValid: Bool {
        choices.count == 4 &&
        (0..<choices.count).contains(correctIndex) &&
        !question.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}
